### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by amit4j**
Following changes were made - 

1. Added a repository overview. 
1. Added description for the repository structure.
1. Reformatted the setup section.
1. Added service endpoint information + sample calls.
1. Added information on how to run the test suite

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,45 @@
...
     - if you're on a Windows PC, execute `vagrant up --provider virtualbox`
-- After this, you will find a vm / docker container running on your system, inside which a docker container running postgres. 
-- To start the flask app, run `./scripts/start.sh` from the project directory. This will start the flask app on port `5000`.
-  - Server logs will be output in the same terminal window.  
-- You can test the service by executing `curl 127.0.0.1/5000` from your shell/terminal. 
+4. After this, you will find a vm / docker container running on your system, inside which a docker container running
+   postgres.
+5. To start the flask app, run `./scripts/start.sh` from the project directory. This will start the flask app on
+   port `5000`.
+
```
Comment (RajathRD) [inline README.md:19]: Add a line regarding running nosetests.

**Comment by RajathRD** [comment]
Hi Amit,

Could you please add a brief description of all the endpoints and the call that needs to be made? [include request header, content].

For example, "To create a recommendation - A PUT request to /recommendations with a json containing _________"

File: README.md (hunk 1 above)
Comment (amit4j) [reply README.md:19]: Sure

File: README.md [hunk 2]
```diff
@@ -43,3 +30,118 @@ tests/                      - test cases package
...
+1. Clone the repo.
+2. Start your Docker Daemon / Virtualbox.
+3. From the project root,
+    - if you're on an M1 mac, execute `vagrant up --provider docker`
+    - if you're on a Windows PC, execute `vagrant up --provider virtualbox`
+4. After this, you will find a vm / docker container running on your system, inside which a docker container running
+   postgres.
+5. To start the flask app, run `./scripts/start.sh` from the project directory. This will start the flask app on
+   port `5000`.
```
[the remark this answers is not in the document]
Comment (amit4j) [reply README.md:44]: The script isn't run from inside the vm. Only from the local.

File: README.md [hunk 3]
```diff
@@ -43,3 +30,118 @@ tests/                      - test cases package
...
+
+- Server logs will be output in the same terminal window.
+- You can test the service by executing `curl 127.0.0.1/5000` from your shell/terminal.
```
[the remark this answers is not in the document]
Comment (amit4j) [reply README.md:47]: Done

File: README.md [hunk 4]
```diff
@@ -43,3 +30,118 @@ tests/                      - test cases package
...
+
+- Endpoint - `GET /recommendations`
+- Returns - returns the list of all recommendations for all products
+- Command
+
+```shell
+curl -X GET \
+  http://localhost:5000/recommendations \
+  -H 'cache-control: no-cache'
+```
```
[the remark this answers is not in the document]
Comment (amit4j) [reply README.md:147]: It's added at the beginning. 

The thought process is to acquaint a new user with the repo, then tell them how to setup + test the service and lastly document the endpoints.

File: README.md [hunk 5]
```diff
@@ -1,13 +1,147 @@
...
-  - Server logs will be output in the same terminal window.  
-- You can test the service by executing `curl 127.0.0.1/5000` from your shell/terminal. 
+4. After this, you will find a vm / docker container running on your system, inside which a docker container running
+   postgres.
+5. To start the flask app, run `./scripts/start.sh` from the project directory. This will start the flask app on
+   port `5000`.
+
+- Server logs will be output in the same terminal window.
+- You can test the service by executing `curl 127.0.0.1/5000` from your shell/terminal.
+
```
[the remark this answers is not in the document]
Comment (amit4j) [reply README.md:60]: Done

**Review by RajathRD — APPROVED** [review]
Great job. 

This Readme contains everything as per the requirements. Thanks Amit.

**Pull request merged**